citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/impl/CrossrefReferenceImportProcessor.java

```diff
@@ -0,0 +1,163 @@
+        itemTypeMapping.put(CrossRefPublication.PROCEEDINGS, ItemType.CONFERENCE_PAPER);
+        itemTypeMapping.put(CrossRefPublication.DATABASE, ItemType.DATABASE);
+        itemTypeMapping.put(CrossRefPublication.STANDARD, ItemType.STATUTE);
+        itemTypeMapping.put(CrossRefPublication.REFERENCE_BOOK, ItemType.DICTIONARY_ENTRY);
+        itemTypeMapping.put(CrossRefPublication.GRANT, ItemType.DOCUMENT);
+        itemTypeMapping.put(CrossRefPublication.DATASET, ItemType.DATABASE);
+        itemTypeMapping.put(CrossRefPublication.BOOK_SERIES, ItemType.BOOK);
+    }
+    
+    public void startImport(KafkaJobMessage message, JobInfo info) {
+        // message = jobToken: eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJKT0IxOTciLCJleHAiOjE2NzgzNDU3NDF9.5Xqh_AoMHcdlatULkCLFtny9pOF_uJ-SRARw0gCybY3h3qHL2mkIIQlk-qTA0Pn0VlhOLuW4FwACHmIdwZVmoA
+        // info = dois: [10.2307/j.ctvcm4h07.67, 10.1515/9780691242507]
+        // null
+        // zotero: byRZjIk2y4e3kay1cnwy3KpB
+        // zoteroId: 9154965
+        logger.info("Starting import for " + info.getDois());
+        
+        ObjectMapper mapper = new ObjectMapper();
+        ArrayNode root = mapper.createArrayNode();
+        int entryCounter = 0;
+
+        sendMessage(null, message.getId(), Status.PROCESSING, ResponseCode.P00);
+        BibEntryIterator bibIterator = null;
+        try {
+            bibIterator = handlerRegistry.handleFile(info, null);
+        } catch (IteratorCreationException e1) {
+            logger.error("Could not create iterator.", e1);
+        }
+        
+        if (bibIterator == null) {
+            sendMessage(null, message.getId(), Status.FAILED, ResponseCode.X30);
+            return;
+        }
+        
+        while (bibIterator.hasNext()) {
+            BibEntry entry = bibIterator.next();
+            if (entry.getArticleType() == null) {
+                // something is wrong with this entry, let's ignore it
+                continue;
+            }
+            ItemType type = itemTypeMapping.get(entry.getArticleType());
+            JsonNode template = zoteroConnector.getTemplate(type);
+            ObjectNode bibNode = generationService.generateJson(template, entry);
+
+            root.add(bibNode);
+            entryCounter++;
+
+            // we can submit max 50 entries to Zotoro
+            if (entryCounter >= 50) {
+                submitEntries(root, info);
+                entryCounter = 0;
+                root = mapper.createArrayNode();
+            }
+
+        }
+        
+        bibIterator.close();
+        
+        ItemCreationResponse response = null;
+        if (entryCounter > 0) {
+            response = submitEntries(root, info);
+        }
+
+        response = response != null ? response : new ItemCreationResponse();
+        sendMessage(response, message.getId(), Status.DONE, ResponseCode.S00);  // giving error 500 as response code mentioned
+    }
```

Comment:
how is this code different than the `FileImportProcessor`?

Reply:
The code in the start of the "startImport" method is bit different. So, have kept them separate.

Reply:
this is a lot of duplicate code. This method should be in the superclass and in line 83 should `getBibIterator()` be called. The rest should then be the same. The logic of how the bibiterator is created will then be in that method.